video.js

```diff
@@ -282,50 +283,54 @@ define(['./teletext', './utils'], function (Teletext, utils) {
             this.dispEnabled |= SCANLINEDISPENABLE;
             this.cursorOn = false;
             this.cursorOff = false;
-
-            if (lastScanline && this.inVertAdjust) {
-                this.endOfFrame();
-                this.inVertAdjust = false;
-            }
-
-            this.lineStartAddr = this.nextLineStartAddr;
         };
 
         this.endOfScanline = function () {
             if (this.scanlineCounter === this.regs[11]) this.cursorOff = true;
 
             this.vpulseCounter = (this.vpulseCounter + 1) & 0x0F;
 
-            if (this.vertAdjustPending) {
-                this.scanlineCounter = 0;
-                this.inVertAdjust = true;
+            var r9Hit = (this.scanlineCounter === this.regs[9]);
```

Comment:
It's a good sign that the code's getting simpler and being phrased more like the control signals we suspect are inside the chip!